base62.go

```diff
@@ -1,81 +1,182 @@
 package ksuid
 
-import "strings"
+const (
+	// lexographic ordering (based on Unicode table) is 0-9A-Za-z
+	base62Characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+	zeroString       = "000000000000000000000000000"
+)
 
-// lexographic ordering (based on Unicode table) is 0-9A-Za-z
-var base62Characters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+// Converts a base 62 byte into the number value that it represents.
+func base62Value(digit byte) byte {
+	switch {
+	case digit >= '0' && digit <= '9':
+		return digit - '0'
+	case digit >= 'A' && digit <= 'Z':
+		return 10 + (digit - 'A')
+	default:
+		return 36 + (digit - 'a')
+	}
+}
 
-func appendBase2Base(dst []byte, src []byte, inBase int, outBase int) []byte {
-	off := len(dst)
-	bs := src[:]
-	bq := [stringEncodedLength]byte{}
+// This function encodes the base 62 representation of the src KSUID in binary
+// form into dst.
+//
+// In order to support a couple of optimizations the function assumes that src
+// is 20 bytes long and dst is 27 bytes long.
+//
+// Any unused bytes in dst will be set to the padding '0' byte.
+func fastEncodeBase62(dst []byte, src []byte) {
+	const srcBase = 4294967296
+	const dstBase = 62
+
+	// Split src into 5 4-byte words, this is where most of the efficiency comes
+	// from because this is a O(N^2) algorithm, and we make N = N / 4 by working
+	// on 32 bits at a time.
+	parts := [5]uint32{
```

Comment:
Can't you generate them with a for loop though? It'll result in more compact code that is easier to read with no performance trade-off.